src/components/organisms/UserProfileModal/UserProfileModal.tsx

```diff
@@ -25,49 +32,53 @@ import Button from "components/atoms/Button";
 
 import "./UserProfileModal.scss";
 
-type PropTypes = {
-  show: boolean;
-  onHide: () => void;
+export interface UserProfileModalProps {
   zIndex?: number;
-  userProfile?: WithId<User>;
-};
+}
 
-const UserProfileModal: React.FunctionComponent<PropTypes> = ({
-  show,
-  onHide,
+export const UserProfileModal: React.FC<UserProfileModalProps> = ({
   zIndex,
-  userProfile,
 }) => {
   const venue = useSelector(currentVenueSelector);
 
   const { user } = useUser();
 
-  const chosenUserId = userProfile?.id;
-
   const { selectRecipientChat } = useChatSidebarControls();
 
+  const {
+    selectedUserProfile,
+    hasSelectedProfile,
+    closeUserProfileModal,
+  } = useProfileModalControls();
+
+  const chosenUserId = selectedUserProfile?.id;
+
   const openChosenUserChat = useCallback(() => {
     if (!chosenUserId) return;
 
     selectRecipientChat(chosenUserId);
     // NOTE: Hide the modal, after the chat is opened;
-    onHide();
-  }, [selectRecipientChat, onHide, chosenUserId]);
+    closeUserProfileModal();
+  }, [selectRecipientChat, closeUserProfileModal, chosenUserId]);
 
-  if (!userProfile || !chosenUserId || !user) {
+  if (!selectedUserProfile || !chosenUserId || !user) {
     return <></>;
   }
```

Comment:
```suggestion
  if (!selectedUserProfile || !chosenUserId || !user) {
    return null;
  }
```

**Further Reading:**

- https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering